src/solvers/dgsem_tree/dg_parallel.jl

```diff
@@ -5,6 +5,14 @@
 @muladd begin
 #! format: noindent
 
+# Function to output the mpi rank for visualization
+function get_element_variables!(element_variables,
+                                mesh::Union{ParallelTreeMesh, ParallelP4estMesh,
+                                            ParallelT8codeMesh})
+    element_variables[:mpi_rank] = mpi_rank()
```

Comment:
Why does this even work? Shouldn't the element variables be an array with one value per element?

Reply:
I guess `element_variables` is already per element, and this is a data structure for this particular element.

Reply:
You are right, @sloede. For the other cell quantities, `element_variables` is indeed structured as an array with one variable per element. Regarding the MPI rank, I managed to get away by storing a single value per MPI rank. This approach works because `trixi2vtk` correctly allocates the necessary data structure for `vtk_celldata` ([here](https://github.com/trixi-framework/Trixi2Vtk.jl/blob/5c3fd0d4976dcdd083a818a7b857f757b7c233d8/src/convert.jl#L162-L164)) and then assigns the element variables from the solution file into `vtk_celldata` ([here](https://github.com/trixi-framework/Trixi2Vtk.jl/blob/5c3fd0d4976dcdd083a818a7b857f757b7c233d8/src/convert.jl#L199-L205)).

Since that worked, I thought it was a nice solution that avoided extra allocations or the creation of cache variables. Do you suggest a different way to implement this?

Reply:
Oh, no. It seems I was wrong. This is only working for parallel HDF5 because it creates the data set explicitly before saving `element_variables`.
https://github.com/trixi-framework/Trixi.jl/blob/df9013ae2f17732501f76e72232d38085b0f74d0/src/callbacks_step/save_solution_dg.jl#L181-L194

However, this implementation is not working for `HDF5.has_parallel() = false`. I will convert this back to a draft and try to solve the issue for `HDF5.has_parallel() = false`.

Reply:
@sloede I just wanted to ping you in case you want to have a look at this solution that I implemented before I resolve the conversation.